Makefile

```diff
@@ -71,35 +74,138 @@ $(GH_PAGES)::
 # update project files
 # --------------------
 
-PHONY += project engines-languages
+PHONY += project engines.languages searx.brand useragents.update
 
-project: searx/data/engines_languages.json
+project: useragents.update engines.languages searx.brand
 
-searx/data/engines_languages.json:  pyenvinstall
-	$(PY_ENV_ACT); python utils/fetch_languages.py
-	mv engines_languages.json searx/data/engines_languages.json
-	mv languages.py searx/languages.py
+engines.languages:  pyenvinstall
+	$(Q)echo "fetch languages .."
+	$(Q)$(PY_ENV_ACT); python utils/fetch_languages.py
+	$(Q)echo "update searx/data/engines_languages.json"
+	$(Q)mv engines_languages.json searx/data/engines_languages.json
+	$(Q)echo "update searx/languages.py"
+	$(Q)mv languages.py searx/languages.py
+
+useragents.update:  pyenvinstall
+	$(Q)echo "Update searx/data/useragents.json with the most recent versions of Firefox."
+	$(Q)$(PY_ENV_ACT); python utils/fetch_firefox_version.py
+
+searx.brand:
+	$(Q)echo "build searx/brand.py"
+	$(Q)echo "GIT_URL = '$(GIT_URL)'"  > searx/brand.py
+	$(Q)echo "ISSUE_URL = 'https://github.com/asciimoo/searx/issues'" >> searx/brand.py
+	$(Q)echo "SEARX_URL = '$(SEARX_URL)'" >> searx/brand.py
+	$(Q)echo "DOCS_URL = '$(DOCS_URL)'" >> searx/brand.py
+	$(Q)echo "PUBLIC_INSTANCES = 'https://searx.space'" >> searx/brand.py
+
+# node / npm
+# ----------
+
+node.env:
+	$(Q)./manage.sh npm_packages
+
+node.clean:
+	$(Q)echo "CLEAN     locally installed npm dependencies"
+	$(Q)rm -rf \
+	  ./node_modules  \
+	  ./package-lock.json \
+	  ./searx/static/themes/oscar/package-lock.json \
+	  ./searx/static/themes/oscar/node_modules \
+	  ./searx/static/themes/simple/package-lock.json \
+	  ./searx/static/themes/simple/node_modules
+
+# build themes
+# ------------
+
+PHONY += themes themes.oscar themes.simple
+themes: themes.oscar themes.simple
+
+themes.oscar:
+	$(Q)echo '[!] Grunt build : oscar theme'
+	$(Q)PATH="$$(npm bin):$$PATH" grunt --gruntfile  "searx/static/themes/oscar/gruntfile.js"
+
+themes.simple:
+	$(Q)echo '[!] Grunt build : simple theme'
+	$(Q)PATH="$$(npm bin):$$PATH" grunt --gruntfile  "searx/static/themes/simple/gruntfile.js"
+
+# build styles
+# ------------
+
+PHONY += styles style.legacy style.courgette style.pixart style.bootstrap
+styles: style.legacy style.courgette style.pixart style.bootstrap
+
+quiet_cmd_lessc = STYLE     $3
+      cmd_lessc = PATH="$$(npm bin):$$PATH" \
+	lessc --clean-css="--s1 --advanced --compatibility=ie9" "searx/static/$2" "searx/static/$3"
+
+style.legacy:
```

Comment:
```style.legacy``` could be renamed ```themes.legacy``` : it actually builds the files for the legacy theme, then  ```themes.legacy``` can be a dependency of ```themes```.

Same for the other styles.*

Debatable: about ```style.bootstrap```, same convention : ```theme.bootstrap``` (even it is more a toolbox for the oscar theme).

So there is no need to add the missing ```make styles``` in the help target.